lib/saluki-components/src/sources/checks/builder/python/python_modules.rs

```diff
@@ -65,23 +120,122 @@ pub mod aggregator {
     }
 
     #[pyfunction]
-    #[pyo3(signature = (name, status, tags, hostname, message=None))]
-    fn submit_service_check(name: String, status: i32, tags: Vec<String>, hostname: String, message: Option<String>) {
-        println!(
+    fn submit_service_check(
+        _class: PyObject, _check_id: String, name: String, status: u8, tags: Vec<String>, hostname: String,
+        message: Option<String>,
+    ) {
+        trace!(
             "submit_service_check called with name: {}, status: {}, tags: {:?}, hostname: {}, message: {:?}",
-            name, status, tags, hostname, message
+            name,
+            status,
+            tags,
+            hostname,
+            message
         );
+
+        if let Ok(service_check_status) = status.try_into() {
+            let mut tag_set = TagSet::default();
+            for tag in tags {
+                tag_set.insert_tag(tag);
+            }
+
+            let service_check = ServiceCheck::new(name, service_check_status)
+                .with_tags(tag_set.into_shared())
+                .with_hostname(if hostname.is_empty() {
+                    None
+                } else {
+                    Some(hostname.into())
+                })
+                .with_message(message.map(Into::into));
+
+            if let Err(e) = try_send_service_check(service_check) {
+                error!("Failed to send service check: {}", e);
+            }
+        } else {
+            error!("Invalid service check status: {}", status);
+        }
```

Comment:
We prefer to unnest conditional structures like this, so that it looks more like:

```rust
let check_status = match status.try_into() {
    Ok(check_status) => check_status,
    _ => {
        error!(check_id, check_status = status, "Check returned invalid/unknown status value.");
        return;
    }
};

// Construct the tags, `ServiceCheck`, etc.
```